apps/engine/lib/engine/ethereum/authority/submitter/external.ex

```diff
@@ -45,8 +45,19 @@ defmodule Engine.Ethereum.Authority.Submitter.External do
     contract = plasma_framework
 
     fn block_root, nonce, gas_price ->
-      apply(module, function, [block_root, nonce, gas_price, contract, external_opts])
-      :ok
+      case apply(module, function, [block_root, nonce, gas_price, contract, external_opts]) do
+        {:ok, %HTTPoison.Response{body: body, status_code: 200}} ->
+          # ENTERPRISE 1
```

Comment:
??? What does this meannn

Reply:
Can `ENTERPRISE 0` and `ENTERPRISE 1` have the same interface? What do you think about wrapping httpoison response and return only tx_hash?

Reply:
no, they can't have the same interface, if by that, you mean the response?
ENTERPRISE 1 case goes through vault
https://github.com/omgnetwork/submit_block_vault/blob/main/lib/submit_block.ex#L36

ENTERPRISE 0 uses eth send raw transaction https://github.com/omgnetwork/submit_block/blob/master/lib/submit_block.ex#L66